jax/experimental/callback.py

```diff
@@ -169,3 +175,70 @@ def process_custom_vjp_call(self, primitive, fun, fwd, bwd, tracers,
     # TODO(sharadmv): don't drop the custom derivative rule
     del primitive, fwd, bwd, out_trees  # Unused.
     return fun.call_wrapped(*tracers)
+
+
+custom_callback_rules: Dict[Any, Any] = {}
+
+def _scan_callback_rule(trace, *tracers, reverse, length, num_consts, num_carry,
+                        jaxpr, linear, unroll):
+  const_tracers, carry_tracers, xs_tracers = split_list(tracers, [num_consts, num_carry])
+  carry_avals, xs_avals = tree_map(lambda x: x.aval, (carry_tracers, xs_tracers))
```

Comment:
```suggestion
  carry_avals, xs_avals = tree_map(op.attrgetter('aval'), (carry_tracers, xs_tracers))
```

Just a stylistic thing I like :) Same elsewhere.

Not worth blocking the merge though!